code/game/machinery/alarm.dm

```diff
@@ -981,44 +967,43 @@ var/global/list/air_alarms = list()
 
 	if(href_list["command"])
 		var/device_id = href_list["id_tag"]
-		switch(href_list["command"])
-			if( "power",
-				"set_external_pressure",
-				"set_internal_pressure",
-				"checks",
-				"co2_scrub",
-				"tox_scrub",
-				"n2o_scrub",
-				"o2_scrub",
-				"n2_scrub",
-				"panic_siphon",
-				"scrubbing",
-				"direction")
-				var/val
-				if(href_list["val"])
-					val=text2num(href_list["val"])
-				else
-					var/newval = input("Enter new value") as num|null
-					if(isnull(newval))
+		var/command = href_list["command"]
+		if(command in XGM.gases)
+			var/val=text2num(href_list["val"])
+			send_signal(device_id, list(command+"_scrub" = val ))
+		else
+			switch(href_list["command"])
+				if( "power",
+					"set_external_pressure",
+					"set_internal_pressure",
+					"checks",
+					"panic_siphon",
+					"scrubbing",
+					"direction")
+					var/val
+					if(href_list["val"])
+						val=text2num(href_list["val"])
+					else
+						var/newval = input("Enter new value") as num|null
+						if(isnull(newval))
+							return 1
+						if(href_list["command"]=="set_external_pressure")
+							if(newval>1000+ONE_ATMOSPHERE)
+								newval = 1000+ONE_ATMOSPHERE
+							if(newval<0)
+								newval = 0
+						val = newval
```

Comment:
I know this is how it was already written, but might as well change it to just use `clamp()`